deploy/helm/elastic-agent/values.yaml

```diff
@@ -450,8 +450,6 @@ agent:
           logs: true
           metrics: true
       providers:
-        kubernetes_leaderelection:
-          enabled: false
```

Comment:
I wonder if keeping this explicitly disabled here isn't better. This provider is enabled by default, and I think it's better to be more explicit when diverging from the agent's default behaviour. It would also make the code in `_helpers.tpl` nicer, as we wouldn't need to handle the edge cases.

Reply:
@cmacknz any opinions on that? I have no issue reverting this to be enabled by default, however, I had the impression we want to minimise the usage of leader election to solely where it is needed.

Reply:
I think the way the Helm chart deploys agent makes this much less necessary (e.g. DaemonSet + Deployment instead of just DaemonSet).

IMO we should turn leaderelection on unless it is necessary for correctness.

Reply:
> IMO we should turn leaderelection on unless it is necessary for correctness.

I have a feeling that here you mean turn leaderelection **off**, right? 🙂

Reply:
I meant to write "we should **not** turn leaderelection on unless it is necessary for correctness"

This would mean it is off by default.